src/core/Track.cpp

```diff
@@ -2653,15 +2653,29 @@ void Track::toggleSolo()
 			{
 				( *it )->m_mutedBeforeSolo = ( *it )->isMuted();
 			}
-			( *it )->setMuted( *it == this ? false : true );
+			// Don't mute AutomationTracks (keep their original state) unless we are on the sololegacybehavior mode
+			if( *it == this ){
+				( *it )->setMuted( false );
+			} else {
+				if( ConfigManager::inst()->value("app","sololegacybehavior").toInt() || ( *it )->type() != AutomationTrack ){
+					( *it )->setMuted( true );
+				}
+			}
```

Comment:
The `else` is guarded with an `if`, so we can merge those blocks. Then since both `if` and `else if` are doing the same thing we should be able to merge with an `||`.
```suggestion
			if( *it == this || ConfigManager::inst()->value("app","sololegacybehavior").toInt() || ( *it )->type() != AutomationTrack ){
				( *it )->setMuted( true );
			}
```

Reply:
This one however can't be merged because the `if` and the `else if` are doing different things (one is setting muted to `false` on the track being soloed and the other setting muted to `true` on the other tracks). I can merge the `else` with the `if` though, it will save two lines at least.

Reply:
> This one however can't be merged because the `if` and the `else if` are doing different things (one is setting muted to `false` on the track being soloed and the other setting muted to `true` on the other tracks). I can merge the `else` with the `if` though, it will save two lines at least.

Oops! You're right. I overlooked the true/false.